lib/infrastructure/user/user_repository.dart

```diff
@@ -0,0 +1,88 @@
+import 'dart:async';
+
+import 'package:collaction_app/domain/user/i_user_repository.dart';
+import 'package:collaction_app/domain/user/user.dart';
+import 'package:firebase_auth/firebase_auth.dart' as firebase_auth;
+import 'package:get_it/get_it.dart';
+import 'package:injectable/injectable.dart';
+import 'package:rxdart/rxdart.dart';
+
+@LazySingleton(as: IUserRepository)
+class UserRepository implements IUserRepository, Disposable {
+  late final firebase_auth.FirebaseAuth _firebaseAuth;
+  final _userSubject = BehaviorSubject<User>.seeded(User.anonymous);
+  late final StreamSubscription _userStreamSubscription;
+
+  UserRepository({required firebase_auth.FirebaseAuth firebaseAuth})
+      : _firebaseAuth = firebaseAuth {
+    _userStreamSubscription = _firebaseAuth
+        .authStateChanges()
+        .map(_firebaseUserToUser)
+        .listen(_userSubject.sink.add);
+  }
+
+  static User _firebaseUserToUser(firebase_auth.User? firebaseUser) {
+    if (firebaseUser == null || firebaseUser.isAnonymous) {
+      return User.anonymous;
+    } else {
+      return User(
+        id: firebaseUser.uid,
+        displayName: firebaseUser.displayName,
+        phoneNumber: firebaseUser.phoneNumber,
+        isPhoneNumberVerified: firebaseUser.phoneNumber != null,
```

Comment:
Is this necessary? Can we do without it?

Reply:
You mean the user Info?
Yeah... I'm also thinking about just providing the idToken and nothing else.

Reply:
Oh I was thinking about the specific line 32, but yeah generally also the user info other than the Id, since any other info will be coming from our own api.